pkg/controllers/edgeconnect/controller.go

```diff
@@ -211,7 +220,8 @@ func (controller *Controller) updateFinalizers(ctx context.Context, edgeConnect
 
 		edgeConnect.ObjectMeta.Finalizers = []string{finalizerName}
 		if err := controller.client.Update(ctx, edgeConnect); err != nil {
-			return err
+			log.Debug("updating finalizers failed", "name", edgeConnect.Name, "namespace", edgeConnect.Namespace, "error", err)
+			return errors.WithStack(err)
```

Comment:
I like the extra context that `Debug` gives, but printing the error there aswell will just cause printing the same error message twice